### PR TITLE
[CARBONDATA-978] Range Filter Evaluation Rectification 

### PR DESCRIPTION
**Pull request by sounakr**
Problem : Range Expression filter evaluation is unable to retreive correct output.

Analysis : Range Filter Expression evaluation is unable to filter out correct values when the filter values are not present in the data. The Binary Search method always returns a value greater than the compared value being passed to it and the value doesn't match in the list. 
For Less Than and Greater than condition the non macthing return value should be adjusted accordingly. 

Fix : Fix the value which is being returned from the binary search method whenever the compared values doesnt match any data in the list.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1773/

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java [hunk 1]
```diff
@@ -431,14 +426,9 @@ private BitSet setFilterdIndexToBitSetWithColumnIndex(
...
-        // index needs to be compared by the filter member if its >= filter then from that
-        // index the bitset will be considered for filtering process.
-        if ((lessThanExp == true) && (ByteUtil.compare(filterValues[1],
-            dimensionColumnDataChunk.getChunkData(dimensionColumnDataChunk.getInvertedIndex(start)))
-            < 0)) {
-          start = start - 1;
-        } else if ((lessThanEqualExp == true) && (ByteUtil.compare(filterValues[1],
+        // In case the start is less than 0, then positive value of start is pointing to the next
+        // value of the searched key. So move to the previous one.
+        if ((ByteUtil.compare(filterValues[1],
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RangeValueFilterExecuterImpl.java]: Please don't use this compare method, it does not compare based on length

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1774/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1776/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**